pkgs/development/libraries/bencode/default.nix

```
@@ -0,0 +1,56 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, catch2
+, cmake
+, expected-lite
+, fmt
+, gsl-lite
+, ninja
+}:
+
+stdenv.mkDerivation rec {
+  pname = "bencode";
+  version = "0.5.0";
+
+  src = fetchFromGitHub {
+    owner = "fbdtemme";
+    repo = "bencode";
+    rev = "v${version}";
+    hash = "sha256-ePbQ6oTumQ+T5TOb+ReaOYa8Q3lvcMBZ6jTMoquTngg=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+  ];
+
+  buildInputs = [
+    catch2
+    expected-lite
+    fmt
+    gsl-lite
+  ];
+
+  postPatch = ''
+    # Disable a test that requires an internet connection.
+    substituteInPlace tests/CMakeLists.txt \
+      --replace "add_subdirectory(cmake_fetch_content)" ""
+  '';
+
+  doCheck = true;
+
+  postInstall = ''
+    rm -rf $out/lib64
+  '';
+
+  meta = with lib; {
+    description = "A header-only C++20 bencode serialization/deserialization library";
+    homepage = "https://github.com/fbdtemme/bencode";
+    changelog = "https://github.com/fbdtemme/bencode/blob/${src.rev}/CHANGELOG.md";
+    license = licenses.mit;
+    maintainers = with maintainers; [ azahi ];
+    platforms = platforms.unix;
+    broken = stdenv.isLinux && stdenv.isAarch64;
```

Darwin is failing too, according to OfBorg

```suggestion
    broken = stdenv.isDarwin || (stdenv.isLinux && stdenv.isAarch64);
```

---

I guess I just forgot to remove this when I was copying the derivation from somewhere.

---

Yes. Looks like it's broken on these targets because of the missing support of C++20. It's possible to test out different `stdenv` versions here and make it work but I don't have access AArch64 or Darwin and I don't feel like waiting for ofborg to build these, so I'll just mark it as broken. Hopefully someone will fix that in the future.